webapp/src/main/java/com/box/l10n/mojito/service/tm/importer/TextUnitBatchImporterService.java

```diff
@@ -92,6 +95,10 @@ public class TextUnitBatchImporterService {
 
   @Autowired MeterRegistry meterRegistry;
 
+  @Autowired SlackWarningMessageSender slackWarningMessageSender;
+
+  @Autowired TMTextUnitRepository tmTextUnitRepository;
```

Comment:
This can be injected into the `SlackWarningMessageSender` instead and the lookup can be handled there